internal/characters/traveler/common/hydro/asc.go

```diff
@@ -0,0 +1,106 @@
+package hydro
+
+import (
+	"github.com/genshinsim/gcsim/internal/common"
+	"github.com/genshinsim/gcsim/pkg/core/attributes"
+	"github.com/genshinsim/gcsim/pkg/core/combat"
+	"github.com/genshinsim/gcsim/pkg/core/geometry"
+	"github.com/genshinsim/gcsim/pkg/core/player"
+	"github.com/genshinsim/gcsim/pkg/core/targets"
+)
+
+const a1ICDKey = "sourcewater-droplet-icd"
+
+// After the Dewdrop fired by the Hold Mode of the Aquacrest Saber hits an opponent, a Sourcewater Droplet will be
+// generated near to the Traveler. If the Traveler picks it up, they will restore 7% HP.
+// 1 Droplet can be created this way every second, and each use of Aquacrest Saber can create 4 Droplets at most.
+func (c *char) makeA1CB() combat.AttackCBFunc {
+	if c.Base.Ascension < 1 {
+		return nil
+	}
+	return func(a combat.AttackCB) {
+		if a.Target.Type() != targets.TargettableEnemy {
+			return
+		}
+		if c.StatusIsActive(a1ICDKey) {
+			return
+		}
+
+		// remove excess droplet
```

Comment:
I believe that you don't remove old droplets, you just stop spawning them after a single E use has made 4 droplets.

Reply:
Yes it should stop spawning once it his 4 in a single e use. In order to do this, you can switch to calling makeA1CB once in E and use the callback for e tick attacks triggered. Then this function would look like this:
```
count := 0
return func(a combat.AttackCB) {
    ...
    if count >= 4 {
        return
    }
}
```